### DISCUSSION TITLE
User-created Theme Support

### DISCUSSION THREAD

**Discussion by agileramblings** [datalust/seq-tickets#1134] [Ideas]
It would be great to have the ability to apply user-generated themes. We have multiple seq environments (prod, lowers, etc) and we use the basic Settings | Instance | Style-Override feature to change the .top-bar for each instance but it would be nice to create more "complete" themes and then share them.

**Comment by nblumhardt** [comment]
Thanks for the suggestion, Dave!

We've been reluctant to do this so far, because we don't want to be penalized for refreshing/changing the UI.

In particular, we're doing some work right now to switch to a new Vue-based UI stack (it's running the Storage screen we introduced in 2020.4) and the way styles work in those components is quite different from what we have in the rest of the codebase.

Just to brainstorm other ways to approach it (since we've had this on the radar a long time), would something as simple as defining/consuming well-known CSS variables for colors be worthwhile? We'd still need to be fairly careful about how we manipulate colors at build time (we use SASS), but it may be easier for us to come up with something workable that way. Any thoughts along those lines?

(CC @larenelg who knows this space much better than I.)

**Comment by agileramblings** [comment]
I don't know that I had any specific approach in mind.  If there was a way to access a "marketplace" of well-known style override fragments and a guide on crafting override fragments, that would certainly be an adequate first step for me. Those style override fragments could certainly take advantage of CSS variables. I was thinking only "color" themes (to help differentiate Seq instances visually) and not any sort of size/location manipulations.

**Comment by nblumhardt** [comment]
Thanks for the follow-up 👍 

I'll move this to a discussion thread, as it's not currently on the short-term roadmap, but definitely still on the radar :-)

**Comment by jeffr1975** [comment]
I would like to change the Messages on the Events screen to monospace font.  Is there an easy way to do that.  I tried setting it for "body".  It worked, but only for the navigation and menus.

**Reply by nblumhardt** [reply]
Hi!

```css
body.route-nav-events .events .rendered-message {
    font-family: monospace;
}
```

in the instance's custom CSS should do it. Let me know if this helps!

**Reply by jeffr1975** [reply]
That definitely worked!  Thanks!
